rust/load/src/workloads.rs

```diff
@@ -0,0 +1,40 @@
+use std::collections::HashMap;
+
+use crate::{Distribution, DocumentQuery, GetQuery, MetadataQuery, QueryQuery, Workload};
+
+pub fn all_workloads() -> HashMap<String, Workload> {
+    HashMap::from_iter([
+        (
+            "get-no-filter".to_string(),
+            Workload::Get(GetQuery {
+                limit: Distribution::Constant(10),
+                metadata: None,
+                document: None,
+            }),
+        ),
+        (
+            "get-document".to_string(),
+            Workload::Get(GetQuery {
+                limit: Distribution::Constant(10),
+                metadata: None,
+                document: Some(DocumentQuery::Raw(serde_json::json!({"$contains": "the"}))),
+            }),
+        ),
+        (
+            "get-metadata".to_string(),
+            Workload::Get(GetQuery {
+                limit: Distribution::Constant(10),
+                metadata: Some(MetadataQuery::Raw(serde_json::json!({"i1": 1000}))),
+                document: None,
+            }),
+        ),
```

Comment:
Depending on how the metadata is generated, an equality check may not be ideal since it is likely to yield empty result (which is not representative of real usages)

Reply:
I plan to make a MetadataQuery variant that scans the different values Evan loaded.

Reply:
Also maybe we could add a few more parameter combinations

Reply:
Yeah, these workloads will eventually compose and (it's along the lines of what @rescrv already mentioned) we should be able to specify how parts of the workload get generated. So, for example, we may want to say "each query should have at least one boolean operator, for each side of the boolean, 25% should have a `$contains` filter, 25% should have a `$not_contains` and 50% have a `$neq`" -- In other words, the components of the actual queries could be programmable and dynamic.